lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -62,75 +65,152 @@ def __init__(self, config):
-            'ontology_PO_obj_name': self.cfg.raw['ontology-plant-ontology-obj-name'],
+            'ontology_wsname': cfg.raw['ontology-workspace-name'],
+            'ontology_GO_obj_name': cfg.raw['ontology-gene-ontology-obj-name'],
+            'ontology_PO_obj_name': cfg.raw['ontology-plant-ontology-obj-name'],
 
             'release': None,
             'genetic_code': 11,
             'generate_ids_if_needed': 0,
             'metadata': {}
         }
 
-    @property
-    def messages(self):
-        return "\n".join(self._messages)
-
-    def refactored_import(self, ctx, params):
-        # 1) validate parameters and extract defaults
-        self.validate_params(params)
-
-        # 2) construct the input directory staging area
-        input_directory = self.stage_input(params)
-
-        # 3) update default params
-        self.default_params.update(params)
-        params = self.default_params
-        self.generate_parents = params.get('generate_missing_genes')
-        self.generate_ids = params.get('generate_ids_if_needed')
-        if params.get('genetic_code'):
-            self.code_table = params['genetic_code']
-
-        # 4) Do the upload
-        files = self._find_input_files(input_directory)
-        consolidated_file = self._join_files_skip_empty_lines(files)
-        genome = self.parse_genbank(consolidated_file, params)
-        if params.get('genetic_code'):
-            genome["genetic_code"] = params['genetic_code']
-
-        result = self.gi.save_one_genome({
-            'workspace': params['workspace_name'],
-            'name': params['genome_name'],
-            'data': genome,
-            "meta": params['metadata'],
-        })
-        ref = f"{result['info'][6]}/{result['info'][0]}/{result['info'][4]}"
-        logging.info(f"Genome saved to {ref}")
 
-        # 5) clear the temp directory
-        shutil.rmtree(input_directory)
-
-        # 6) return the result
-        info = result['info']
-        details = {
-            'genome_ref': ref,
-            'genome_info': info
-        }
+class GenbankToGenome:
+    def __init__(self, config):
+        self.cfg = config
+        self.gi = GenomeInterface(config)
+        self.dfu = DataFileUtil(config.callbackURL)
+        self.aUtil = AssemblyUtil(config.callbackURL)
+        self.ws = Workspace(config.workspaceURL)
+        self.re_api_url = config.re_api_url
+        yml_text = open('/kb/module/kbase.yml').read()
+        self.version = re.search("module-version:\n\W+(.+)\n", yml_text).group(1)
 
+    def import_genbank(self, params):
+        print('validating parameters')
+        mass_params = self._set_up_single_params(params)
+        return self._import_genbank_mass(mass_params)[0]
+
+    def import_genbank_mass(self, params):
+        print('validating parameters')
+        self._validate_mass_params(params)
+        return self._import_genbank_mass(params)
+
+    def _set_up_single_params(self, params):
+        inputs = dict(params)
```

Comment:
Is this to avoid function side effects? If so can you add a comment to that effect?

Reply:
👍

Reply:
:+1: